ui/src/app-bar/app-bar.tsx

```diff
@@ -44,6 +44,24 @@ export const AppBar: React.FC = () => {
     const pvDocumentCounts = useLoader(() => getPVDocumentCount(), [changeTimes.pvDocument]);
     const exclamationPointVisibility = !!pvDocumentCounts && pvDocumentCounts?.suggested > 0;
 
+    const selectedPublicationId = useTrackLayoutAppSelector(
```

Comment:
Aiempi julkaisuloki käytti tätä selectoria/redux-storea. Pitäisikö etusivua varten luoda ihan oma slice/reducer/selector jne? Sinänsä rakenteen puolesta vähän hähmeä, sillä app-bar kaipaisi sitten sen slicen sisältöä.

Reply:
Hyvä nosto. Tuossa aiemmassa ratkaisussahan on tosiaan useampi juttu vinksallaan (vaikka taisinkin itse sen aikanaan koodata.) `TrackLayout`:han viittaa meillä karttanäkymään, ja etusivu ei ole millään tavalla karttanäkymään sidonnainen asia -> tämä puoltaisi sitä, että tämä pois täältä ja etusivulle tosiaan oma store/selector/mikälie.

Toisaalta kumminkin se syy miksi `TrackLayoutStore`:a tässä käytetään on se, että meillä kaikki valitut asiat asuu sen alla. Täällä kumminkin selkeästi valitaan jokin julkaisu katseltavaksi. Intuitio sanoisi että meillä kannattaisi olla erilliset redux-storet kaikille päänäkymille (kartalle, inframodelille ja tietotuotteillehan on jo,) joissa kunkin näkymän sisäiset selectionit pääsevät elämään omaa elämäänsä. Tällöin tosin nykyinen `Selection` kannattaisi varmaankin nimetä uudestaan joksikin tyyliin `MapSelection`:iksi sekä tietotyyppi- että store-tasolla, jotta se kuvaisi paremmin sitä että sen on tarkoitus sisältää tavaraa ainoastaan karttanäkymään liittyen.

Mutta nuo voi kyllä hoitaa jollain omalla tiketillä, ei tarvi tässä yhteydessä lähteä refakkaamaan koko maailmaa.

Reply:
👍 Tiketeerattu @ GVT-2369